### DISCUSSION TITLE
TFB benchmark suite network - host or tfb?

### DISCUSSION THREAD

**Discussion by VStepany** [TechEmpower/FrameworkBenchmarks#9274] [Q&A]
I'm having issues when trying to perform TFB benchmarking runs.

The Benchmarking page suggest the suite should be run using the following Docker command, which uses **"host" network** AND **client, database/ app servers IPs** (which could be replaced by 127.0.1.1 is run locally)
`docker run -e USER_ID=$(id -u) --network=host --mount type=bind,source=/home/techempower/FrameworkBenchmarks,target=/FrameworkBenchmarks techempower/tfb --server-host 127.0.1.1 --database-host 127.0.1.1 --client-host 127.0.1.1 --network-mode host` 

The Development section suggests using **"tfb"** script instead which runs a Docker commands as well, but with few different settings:
- it uses "tfb" newtork for Docker containers to communicate with each other
- it doesn't specify database/client/app servers IPs

I tried running few ASP.NET tests (aspnetcore, aspnetcore-mysql, aspnetcore-mvc).

With "Benchmarking" command they all FAIL because of not being able to access "URL http://127.0.1.1:8080".
With the "Development" "tfb" script the same tests run smoothly, while accessing "URL http://tfb-server:8080". However, this is a development mode, without taking into consideration the 3 machines setup.

The instructions are confusing and won't elaborate on the steps to take to run TFB in a real world 3 machine setup. 

I'd appreciate any input on hoe to set up the right Docker command to run the TFB in benchmarking mode with 3 machines for database, client, app server roles.

**Comment by volyrique** [comment]
Out of curiosity I tried your benchmarking command line and it worked fine for me. The only things that I modified were that I changed the path from `/home/techempower/FrameworkBenchmarks` to whatever I used on my system and that I added `--test aspnetcore aspnetcore-mysql aspnetcore-mvc` to the command line.

**Comment by joanhey** [comment]
If you want to try the bench with 3 machines, the firs think is change the IP for all the host !!

`docker run -e USER_ID=$(id -u) --network=host --mount 
type=bind,source=/home/techempower/FrameworkBenchmarks,target=/FrameworkBenchmarks techempower/tfb --server-host 127.0.1.1 --database-host 127.0.1.1 --client-host 127.0.1.1 --network-mode host`

Your benchmark point all servers to the same IP 127.0.1.1
